### PR TITLE
Add 'Using Docker' section to readme

### PR DESCRIPTION
**Pull request by jpsim**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -108,6 +108,26 @@ adb push tests/tests /data/local/tmp
...
+
+The `Dockerfile` included in this repo will provision a Docker image suitable
+for building and running tests for both the Linux and Android platforms.
+
+```
+# Build Docker image from Dockerfile
+docker build -t "objectstore" .
+# Run bash interactively from the built Docker image,
+# mounting the current directory
+docker run -it -v `pwd`:`pwd` -w `pwd` objectstore bash
```
Comment (emanuelez) [inline README.md]: I'd prefer something like `docker run --rm -it -v $(pwd):/tmp -w /tmp objectstore bash`

File: README.md [hunk 2]
```diff
@@ -108,6 +108,26 @@ adb push tests/tests /data/local/tmp
...
+# Build the object store for Linux and run tests
+> cmake .
```
Comment (emanuelez) [inline README.md]: Let's suggest a more idiomatic use of CMake

```
mkdir build
cd build
cmake ..
make
make run-tests
```

Comment (bdash) [inline README.md]: Elsewhere in the object store README we assume that you're building in the root of the source directory. Using a different directory also works, but makes the instructions more convoluted than they need to be.

File: README.md (hunk 1 above)
Comment (jpsim) [inline README.md]: Done in e67d4d0.

File: README.md (hunk 2 above)
Comment (jpsim) [inline README.md]: I stuck with the instructions already in the readme. If you think they can be improved, I suggest you make a PR changing both the main instructions along with their use in this section, so we stay consistent.

**Comment by jpsim** [comment]
Thanks for the feedback y'all!

**Pull request merged**